graphql/apollo.ts

```diff
@@ -28,7 +28,28 @@ export default function getApolloClient ( serverSideContext?: GetServerSideProps
 	// Server-side / static: Return a new instance every time.
 	if ( isServerSide ) {
 		// @ts-expect-error: Express locals are not defined on Next.js request.
-		const { requestContext } = serverSideContext?.res?.locals || {};
+		let { requestContext } = serverSideContext?.res?.locals || {};
+
+		// @ts-expect-error: res may not be defined
```

Comment:
I think we can avoid this error with type narrowing. The type for `Response` should be available, it's only `res.locals` that isn't available since it's basically a vehicle for custom data. (We could probably declare a type for that, too.)

Reply:
I had tried that, but it didn't work as `serverSideContext` can come from either `GetServerSidePropsContext` or `GetStaticPropsContext`. The former has res as `ServerResponse` but the latter does not. That's why we need the `?` from what I understood. I could use some workarounds to get the type, but that's going to impact performance. 

That's why left it using @ts-expect-error approach instead

Reply:
Here's an example of [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) ("type predicates") that can help you navigate this situation that shouldn't impact performance:

```
function isServerSidePropsContext( context: GetServerSidePropsContext | GetStaticPropsContext ): context is GetServerSidePropsContext {
	return ( context as GetServerSidePropsContext ).res !== undefined;
}
```

Reply:
Incorporated this in the common method, so type narrowing is in!